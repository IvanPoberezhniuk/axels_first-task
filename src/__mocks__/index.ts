import { Comment, Image, ImageDetails } from '../redux/ducks/images';

export const mockedImage: Image = {
  id: 222,
  url: '/fakeUri/images'
};

export const mockedComment: Comment = {
  id: 222,
  text: 'test',
  date: new Date().getSeconds()
};

export const mockedImagesDetails: ImageDetails = {
  id: 222,
  url: '/fakeUri/images',
  comments: [{
    id: 222,
    text: 'test',
    date: new Date().getSeconds()
  }, {
    id: 222,
    text: 'test',
    date: new Date().getSeconds()
  }]
};

export const mockedStore = {
  imagesStore: {
    loading: false,
    images: [],
    imageDetails: {
      id: undefined,
      url: undefined,
      comments: []
    }
  }
};
