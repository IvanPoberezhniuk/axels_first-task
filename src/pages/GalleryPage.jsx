import React, {useEffect, useState} from 'react';
import {Col, Container, Row} from 'react-bootstrap';
import {useDispatch, useSelector} from 'react-redux';

import {ImageCard, ImageModal} from '../components';

import {Link, useLocation} from 'react-router-dom';
import {fetchImages} from '../store/modules/images';

const GalleryPage = () => {
  const [modalShow, setModalShow] = useState(false);
  const [imgId, setImgId] = useState(null);
  const location = useLocation();
  //redux
  const dispatch = useDispatch();
  const imagesList = useSelector(store => store.imagesStore.images);

  useEffect(() => {
    dispatch(fetchImages());
  }, []);

  const setModal = (id) => {
    setModalShow(true);
    setImgId(id);
  };

  return (
    <div>
      <Container>
        <Row>
          {imagesList.map((img) => (
            <Col xs={12} md={6} lg={4} xl={4} key={img.id}>
              <Link to={{pathname: `/gallery/${img.id}`, state: {background: location}}}>
                <ImageCard url={img.url} showModal={() => setModal(img.id)}/>
              </Link>
            </Col>)
          )}
        </Row>
      </Container>
      {imgId && <ImageModal show={modalShow} imgId={imgId} onHide={() => {
        setModalShow(false);
      }}/>}
    </div>
  );
};

GalleryPage.propTypes = {};

export default GalleryPage;
