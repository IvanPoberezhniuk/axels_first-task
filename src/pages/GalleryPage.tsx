import React, { useEffect } from 'react';
import { Col, Container, Row } from 'react-bootstrap';
import { RootStateOrAny, useDispatch, useSelector } from 'react-redux';
import { Link, useLocation } from 'react-router-dom';

import { ImageCard } from '../components';

import { fetchImages, Image } from '../redux/ducks/images';
import { AppDispatch } from '../redux/configureStore';

const GalleryPage = () => {
  const location = useLocation();
  const dispatch: AppDispatch = useDispatch();
  const imagesList: Array<Image> = useSelector((store: RootStateOrAny) => store.imagesStore.images);

  useEffect(() => {
    dispatch(fetchImages());
  }, [dispatch]);

  return (
    <div>
      <Container>
        <Row>
          {imagesList.map((img: Image) => (
            <Col xs={12} md={6} lg={4} xl={4} key={img.id}>
              <Link
                to={{
                  pathname: `/gallery/${img.id}`,
                  state: { background: location }
                }}
              >
                <ImageCard url={img.url} />
              </Link>
            </Col>
          ))}
        </Row>
      </Container>
    </div>
  );
};

export default GalleryPage;
