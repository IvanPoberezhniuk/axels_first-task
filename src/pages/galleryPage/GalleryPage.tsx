import React, { useEffect } from 'react';
import { Col, Container, Row } from 'react-bootstrap';
import { useDispatch, useSelector } from 'react-redux';
import { Link, useLocation } from 'react-router-dom';
import { ImageCard } from '../../components';

import { fetchImages, Image } from '../../redux/ducks/images';
import { AppDispatch } from '../../redux/configureStore';
import Selectors from '../../redux/selectors';

const GalleryPage = (props: any) => {
  const location = useLocation();
  const dispatch: AppDispatch = useDispatch();
  const imagesList: Array<Image> = useSelector(Selectors.images);

  useEffect(() => {
    dispatch(fetchImages());
  }, [dispatch, props]);

  return (
    <div>
      <Container>
        <Row>
          {imagesList && imagesList.map((img: Image) => (
            <Col xs={12} md={6} lg={4} xl={4} key={img.id}>
              <Link
                to={{
                  pathname: `/gallery/${img.id}`,
                  state: { background: location }
                }}
              >
                <ImageCard url={img.url} />
              </Link>
            </Col>
          ))}
        </Row>
      </Container>
    </div>
  );
};

export default GalleryPage;
