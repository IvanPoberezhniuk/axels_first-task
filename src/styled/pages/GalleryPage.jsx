import React, {useEffect, useState} from 'react';
import ImageCard from '../components/ImageCard';
import {getImages} from '../api/httpRequests';
import {Col, Container, Row} from 'react-bootstrap';
import ImageModal from '../components/ImageModal';

const GalleryPage = () => {
  const [imgList, setImgList] = useState([]);
  const [modalShow, setModalShow] = useState(false);
  const [imgId, setImgId] = useState(null);

  useEffect(() => {
    getImages().then(data => setImgList(data));
  }, []);

  const setModal = (imgId) => {
    setModalShow(true);
    setImgId(imgId);
  };

  return (
    <div>
      <Container>
        <Row>
          {imgList.map((img) => (
            <Col xs={12} md={6} lg={4} key={img.id}>
              <ImageCard url={img.url} showModal={() => setModal(img.id)}/>
            </Col>)
          )}
        </Row>
      </Container>
      {imgId && <ImageModal show={modalShow} imgId={imgId} onHide={() => setModalShow(false)}/>}
    </div>
  );
};

GalleryPage.propTypes = {};

export default GalleryPage;
