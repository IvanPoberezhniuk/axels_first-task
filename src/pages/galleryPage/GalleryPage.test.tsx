import { shallow, ShallowWrapper } from 'enzyme';
import { Link } from 'react-router-dom';

import GalleryPage from './GalleryPage';

import { mockedImage } from '../../__mocks__';

let component: ShallowWrapper;
const mockUseDispatch = jest.fn();
const mockUseLocation = jest.fn();
const setUp = () => shallow(<GalleryPage />);

beforeEach(async () => {
  component = await setUp();
});

afterEach(() => {
  jest.resetAllMocks();
});

jest.mock('react-redux', () => ({
  ...jest.requireActual<object>('react-redux'),
  useDispatch: () => mockUseDispatch,
  useSelector: () => ([mockedImage, mockedImage])
}));

jest.mock('react-router-dom', () => ({
  ...jest.requireActual<object>('react-router-dom'),
  useLocation: () => mockUseLocation
}));

describe('GalleryPage component', () => {
  it('should take snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('should render all images from list', () => {
    expect(component.find(Link).children()).toHaveLength(2);
  });
});


