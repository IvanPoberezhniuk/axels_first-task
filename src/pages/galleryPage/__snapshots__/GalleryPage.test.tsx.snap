// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryPage Component Snapshot SNAPSHOT 1`] = `
<div>
  <Container
    fluid={false}
  >
    <Row
      noGutters={false}
    >
      <Col
        key="222"
        lg={4}
        md={6}
        xl={4}
        xs={12}
      >
        <Link
          to={
            Object {
              "pathname": "/gallery/222",
              "state": Object {
                "background": [MockFunction],
              },
            }
          }
        >
          <ImageCard
            url="/fakeUri/images"
          />
        </Link>
      </Col>
      <Col
        key="222"
        lg={4}
        md={6}
        xl={4}
        xs={12}
      >
        <Link
          to={
            Object {
              "pathname": "/gallery/222",
              "state": Object {
                "background": [MockFunction],
              },
            }
          }
        >
          <ImageCard
            url="/fakeUri/images"
          />
        </Link>
      </Col>
    </Row>
  </Container>
</div>
`;
